transcom/mymove
# B 18800 county with address response int

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-18800)

Adding county as a nullable field to the existing Address definition in order to return the county for responses. For any address returned in a response, this should now include the county.

This was added to the existing Address definition rather than defining a new response definition due to simplicity and international moves not having a county.

This does technically allow the county to be provided in the request body, but there is no way to input the county in the UI. Even when a county is provided in the request, and if the county didn't match the zip for some reason, _we return the correct county_ that we calculated based on the provided zip in the response.

The end goal is to always return the county in a response containing an address.

Changed as part of this PR:
- API responses return the county in their response
- Added county to `adminmessages` `address` struct
- Added county to `supportmessages` `address` struct
- Tried to include county in existing tests where it made sense

To test:
Test out the internal API, GHC API, prime API, and (if you can think of an endpoint that returns an address) the admin API. I used swagger for testing.

Internal API:
1. Log in as a service member. I logged in with the email multiplemoves@PPM.com which is test service member Tori Ross. Make a request to get/service_members/{serviceMemberId} and verify that the response includes the counties for the service member residential address and backup address.

GHC API:
1. Log in as an office user
2. Some example endpoints to test and make sure you get county in response:
- GET move/{locator}
- GET orders/{orderID}
3. Also test that it isn't required on a request but you still see county in response:
- POST /customer

Prime API:
1. Log in as office user
2. Some example endpoints to test and make sure you get county in response
- PUT /mto-shipments/{mtoShipmentID}/addresses/{addressID}
4. Test that even if you input a county that does not match the zip you put in the request, that you still see the correct county for your provided zip returned in the response:
Using a county of "INCORRECT COUNTY" and a zip of 67492 in the request:
![image](https://github.com/transcom/mymove/assets/147537467/1267358a-61b7-4d03-bfef-2c9ce322319e)

Returns the correct county in response (for 67492, that would be Dickinson):
 
![image](https://github.com/transcom/mymove/assets/147537467/6640d3a8-acee-4387-a183-168526ecb7d5)
